### PR TITLE
Solution of my implementation  MachineServise Int

### PR DESCRIPTION
**Pull request by dima-semeniuk**

### CODE REVIEW

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 1]
```diff
@@ -1,7 +1,41 @@
...
+    @Override
+    public void fill(List<? super Machine> machines, Machine value) {
+        for (int i = 0; i < machines.size(); i++) {
+            machines.set(i, value);
+        }
+    }
+
+    @Override
+    public void startWorking(List<? extends Machine> machines) {
+        for (int i = 0; i < machines.size(); i++) {
```
Comment (yurii-tomashevskyi) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:37]: I think beter use for each loop

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 2]
```diff
@@ -1,7 +1,41 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<Machine> machines = new ArrayList<>();
+        if (type == Bulldozer.class) {
+            machines.addAll(new BulldozerProducer().get());
+            return machines;
```
Comment (AlexRozv) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:16]: why do you have to fill your list of machines and then return it if you can just return new BulldozerProducer().get()

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 3]
```diff
@@ -1,7 +1,41 @@
...
+        }
+        if (type == Excavator.class) {
+            machines.addAll(new ExcavatorProducer().get());
+            return machines;
+        } else if (type == Truck.class) {
+            machines.addAll(new TruckProducer().get());
+            return machines;
+        }
+        return machines;
+    }
```
Comment (AlexRozv) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:26]: also, you can remove machines list completely and just return empty list here

**Review by AlexRozv — CHANGES_REQUESTED** [review]

**Review by AlexRozv — APPROVED** [review]
well done ✅

**Pull request closed without merging**